File: test-fixtures/typescript-test/src/services.ts
import { User, UserService, UserRepository } from './types';

export class UserManager {
    constructor(
        private service: UserService,
        private repository: UserRepository
    ) {}
    
    async loadUser(id: string): Promise<User | null> {
        const cached = this.service.getUser(id);
        if (cached) return cached;
        
        return await this.repository.findById(id);
    }
}

File: test-fixtures/go-test/services.go
package main

// UserManager coordinates user operations
type UserManager struct {
service    *UserService
repository UserRepository
}

// NewUserManager creates a new UserManager
func NewUserManager(service *UserService, repo UserRepository) *UserManager {
return &UserManager{
service:    service,
repository: repo,
}
}

// LoadUser loads a user from cache or repository
func (m *UserManager) LoadUser(id string) (*User, error) {
if user, ok := m.service.GetUser(id); ok {
return user, nil
}

return m.repository.FindByID(id)
}
